testing/test_assertrewrite.py

```diff
@@ -65,13 +65,15 @@ class TestAssertionRewrite(object):
     def test_place_initial_imports(self):
         s = """'Doc string'\nother = stuff"""
         m = rewrite(s)
-        assert isinstance(m.body[0], ast.Expr)
-        assert isinstance(m.body[0].value, ast.Str)
-        for imp in m.body[1:3]:
+        if sys.version_info < (3, 7):
```

Comment:
I'm trying to understand this, why remove the docstring from the beginning of the AST in the test? If it should not be there, then perhaps we should `assert` what we expect to be there in `>=3.7`?

Reply:
Module docstring is now (in 3.7) a part of Module node, it's not in the body. So I check if there's docstring expression in Python prior to 3.7 and then remove it so the following body items have the same indexes on both Pythons. This way, for example, `m.body[0:2]` is a slice of imports.

I didn't like this approach. So I split this test in 3: one without docstrings that works on both versions, one with docstrings that's skipped prior to 3.7, and one with docstrings for versions after 3.7. This looked quite ugly as well to me, so I reverted to this variant.

Reply:
> Module docstring is now (in 3.7) a part of Module node, it's not in the body. So I check if there's docstring expression in Python prior to 3.7 and then remove it so the following body items have the same indexes on both Pythons. This way, for example, m.body[0:2] is a slice of imports.

@Perlence thanks for the detailed explanation, everything is clear to me now.